### PR TITLE
Arm64 Sos fixes for

### PR DESCRIPTION
**Pull request by ramarag**
!verify heap
!u
!u -gcinfo

@rahku @dotnet/arm64-contrib  @dotnet/debugger-team  PTAL

### CODE REVIEW

**Comment by ramarag** [comment]
Fixes #6514 #5780 #5778 #5705

File: src/ToolBox/SOS/Strike/gcroot.cpp [hunk 1]
```diff
@@ -1320,11 +1320,11 @@ void PrintNotReachableInRange(TADDR rngStart, TADDR rngEnd, BOOL bExcludeReadyFo
 // The value of card_size is determined empirically according to the average size of an object
 // In the code we also rely on the assumption that one card_table entry (DWORD) covers an entire os page
 //
-#if defined (_TARGET_AMD64_)
+#if defined (_TARGET_WIN64_)
```
Comment (ramarag) [inline src/ToolBox/SOS/Strike/gcroot.cpp]: @billwert fix for !verifyheap

**Comment by noahfalk** [comment]
@mikem8361

**Comment by mikem8361** [comment]
LGTM

**Comment by rahku** [comment]
!gcinfo still does not emit all the information. Eg. liveness of register/stack slots seem to be missing. Also it always says prolog size is 0 which is not correct.
0:000> !gcinfo 00007ffa`e324e3a0
entry point 00007ffae324e330
preJIT generated code
GC info 00007ffae3608bfc
Pointer table:
Prolog size: 0
Security object: <none>
GS cookie: <none>
PSPSym: <none>
Generics inst context: <none>
PSP slot: <none>
GenericInst slot: <none>
Varargs: 0
Frame pointer: Fp
Wants Report Only Leaf: 0
Size of parameter area: 0
Return Kind: Scalar
Code size: c0
00000070 is a safepoint: 
00000094 is a safepoint:

**Comment by rahku** [comment]
ignore the liveness reporting issue ....the method i was looking does not have live registers. Liveness is being reported correctly.

**Comment by ramarag** [comment]
For most methods even in desktop the prolog size is 0.
I was able to find only 1 method System.IO.UnmanagedMemoryStream.Initialize(Byte*, Int64, Int64, System.IO.FileAccess, Boolean) which had non zero prolog size, Arm64 produces a sane looking value for  it

Will open another issue if the prolog size reported is incorrect 

@BruceForstall @mikem8361 can you tell me what prolog size reported in !gcinfo actually translates into ?

**Pull request merged**